src/resolvers/shipment/index.ts

```diff
@@ -0,0 +1,82 @@
+import { UserInputError, ApolloError } from 'apollo-server'
+
+import {
+  QueryResolvers,
+  MutationResolvers,
+  ShipmentResolvers,
+  ShipmentInput,
+} from '../../server-internal-types'
+import { sequelize } from '../../sequelize'
+import Shipment from '../../models/shipment'
+import Group from '../../models/group'
+
+const shipmentRepository = sequelize.getRepository(Shipment)
+const groupRepository = sequelize.getRepository(Group)
+
+// Shipment query resolvers
+const listShipments: QueryResolvers['listShipments'] = async () => {
+  return shipmentRepository.findAll()
+}
+
+// Shipment mutation resolvers
+const addShipment: MutationResolvers['addShipment'] = async (
+  _parent,
+  { input },
+  _context,
+) => {
+  if (
+    !input.shippingRoute ||
+    !input.labelYear ||
+    !input.labelMonth ||
+    !input.sendingHubId ||
+    !input.receivingHubId ||
+    !input.status
+  ) {
+    throw new UserInputError('Shipment arguments invalid', {
+      invalidArgs: Object.keys(input).filter(
+        (key) => !input[key as keyof ShipmentInput],
+      ),
+    })
+  }
+
+  const sendingHub = await groupRepository.findByPk(input.sendingHubId)
+  if (!sendingHub) {
+    throw new ApolloError('Sending hub not found')
+  }
+
+  const receivingHub = await groupRepository.findByPk(input.sendingHubId)
+  if (!receivingHub) {
+    throw new ApolloError('Receiving hub not found')
+  }
```

Comment:
suggestion (non-blocking): I think retrieving the sending and receiving hubs could be done concurrently rather than in strict serial order by doing this:

```typescript
  const sendingHubPromise = groupRepository.findByPk(input.sendingHubId)
  const receivingHubPromise = groupRepository.findByPk(input.sendingHubId)
  
  
  const sendingHub = await sendingHubPromise
  if (!sendingHub) {
    throw new ApolloError('Sending hub not found')
  }
  
  const receivingHub = await receivingHubPromise
  if (!receivingHub) {
    throw new ApolloError('Receiving hub not found')
  } 
```

Reply:
Will make this change. Much better.